resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1618,9 +1618,9 @@ private[spark] object Client extends Logging {
       return false
     }
 
-    val srcAuthority = srcUri.getAuthority()
-    val dstAuthority = dstUri.getAuthority()
-    if (srcAuthority != null && !srcAuthority.equalsIgnoreCase(dstAuthority)) {
+    val srcUserInfo = srcUri.getUserInfo()
+    val dstUserInfo = dstUri.getUserInfo()
+    if (srcUserInfo != null && !srcUserInfo.equalsIgnoreCase(dstUserInfo)) {
```

Comment:
This does not handle the case of `srcUserInfo == null && dstUserInfo != null`

Reply:
srcUserInfo == null && dstUserInfo ! = null is fine in this case。equalsIgnoreCase I'm not sure if this is correct, but I'm going to follow the same logic as before。But the use of equal is more rigorous。

Reply:
We don't need to do this adaptation for the user!

Reply:
Last pr discussion：https://github.com/apache/spark/pull/19885#discussion_r154819072

Reply:
`getAuthority` is not null since it includes the host - `getUserInfo` can be `null`